plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -144,29 +183,38 @@ processTerm
     :: Inlining tyname name uni fun a m
     => Term tyname name uni fun a
     -> m (Term tyname name uni fun a)
-processTerm = \case
-    v@(Var _ n) -> do
+processTerm = handleTerm <=< traverseOf termSubtypes applySubstitution where
+    handleTerm = \case
+        v@(Var _ n) -> do
+            subst <- get
+            pure $ case lookupTerm n subst of
+                -- Not substituted for, leave it as it is
+                Nothing       -> v
+                -- Already processed term, just put it in, don't do any further optimization here.
+                -- See Note [Inlining approach and 'Secrets of the GHC Inliner']
+                -- NOTE:  We need to rename it to avoid variable capture
```

Comment:
Missing call to rename?

Reply:
I am not sure that we actually need this one, but I think I will leave it just in case.